docs/contributing/contributing-rules.md

```diff
@@ -1,151 +1,415 @@
 
-For an example of a good rule message, see: [this rule for Django's mark_safe](https://semgrep.dev/r?q=python.django.security.audit.avoid-mark-safe.avoid-mark-safe).
+Use the YAML multiline string operator `>-` when rule messages span multiple lines. This presents the best-looking rule message on the command line without having to worry about line wrapping or escaping the quote or using backslash.
 
-Use the YAML multiline string operator `>-` when rule messages span multiple lines. This presents the best-looking rule message on the command-line without having to worry about line wrapping or escaping the quote or backslash characters.
+For an example of a good rule message, see: [this rule for Django's mark_safe](https://semgrep.dev/r?q=python.django.security.audit.avoid-mark-safe.avoid-mark-safe).
 
-:::info
-`mark_safe()` is used to mark a string as *safe* for HTML output. This disables escaping and may expose the content to XSS attacks. Instead, use `django.utils.html.format_html()` to build HTML for rendering.
+:::info Rule message example
+`mark_safe()` is used to mark a string as safe for HTML output. This disables escaping and may expose the content to XSS attacks. Instead, use `django.utils.html.format_html()` to build HTML for rendering.
 :::
 
-### Rule quality checker
+### Including additional details with rule metadata
 
-When you contribute rules to the rules repository, our quality checkers (linters) evaluate if the rule conforms to r2c standards. The `semgrep-rule-lints` job runs linters on a new rule to check for mistakes, performance problems, and best practices for submitting to the Semgrep rules repository. To improve your rule writing, use Semgrep itself to [scan semgrep-rules](https://r2c.dev/blog/2021/how-we-made-semgrep-rules-run-on-semgrep-rules/).
+Rules in the Semgrep Registry require specific metadata fields that ensure consistency across the ecosystem in both Semgrep App and Semgrep CLI.
+
+Nest these metadata under the `metadata` key. The following metadata are required:
+
+<table>
+  <thead><tr>
+   <th>Required metadata field</th>
+   <th>Possible values</th>
+   <th>Example use</th>
+  </tr></thead>
+  <tbody>
+  <tr>
+   <td>CWE</td>
+   <td>A <a href="https://cwe.mitre.org/index.html">Comment Weakness Enumeration (CWE)</a>.</td>
+   <td><pre>cwe: "CWE-502: Deserialization of Untrusted Data"</pre></td>
+  </tr>
+  <tr>
+   <td>Confidence</td>
+   <td><code>HIGH</code>, <code>MEDIUM</code>, <code>LOW</code></td>
+   <td><pre>confidence: MEDIUM</pre></td>
+  </tr>
+  <tr>
+   <td>Likelihood</td>
+   <td><code>HIGH</code>, <code>MEDIUM</code>, <code>LOW</code></td>
+   <td><pre>likelihood: MEDIUM</pre></td>
+  </tr>
+  <tr>
+   <td>Impact</td>
+   <td><code>HIGH</code>, <code>MEDIUM</code>, <code>LOW</code></td>
+   <td><pre>impact: HIGH</pre></td>
+  </tr>
+  <tr>
+   <td>Subcategory</td>
+   <td><code>vuln</code>, <code>audit</code>, <code>guardrail</code></td>
+   <td>
+    <pre>
+        subcategory:<br />
```

Comment:
@LewisArdern We have a description of the Subcategory, but we have not included a description of the Category.